source/Nevermore/Transient/SqlDatabaseTransientErrorDetectionStrategy.cs

```diff
@@ -198,18 +198,27 @@ sealed class SqlDatabaseTransientErrorDetectionStrategy : ITransientErrorDetecti
 
         public bool IsTransient(Exception ex)
         {
-            if (ex is TimeoutException) return true;
+            return ex switch
+            {
+                TimeoutException => true,
+                InvalidOperationException invalidOperationException => IsPooledConnectTimeout(invalidOperationException),
+                SqlException sqlException => IsTransientSqlException(sqlException),
+                _ => false
+            };
+        }
 
-            var sqlException = ex as SqlException;
-            if (sqlException == null) return false;
+        static bool IsPooledConnectTimeout(InvalidOperationException exception)
+             => exception.Message.Contains("The timeout period elapsed prior to obtaining a connection from the pool.");
```

Comment:
This is not ideal, but there is no error code or similar attached to the exception raised. Type and message string is all we have to go on ([source](https://github.com/dotnet/SqlClient/blob/main/src/Microsoft.Data.SqlClient/src/Microsoft/Data/Common/AdapterUtil.cs#L1141-L1142)).

Reply:
I am curious if this will break if the system language isn't English? and do we support that in the first place?

(I don't think it's a showstopper anyway, it will just go back to the current behaviour)

Reply:
Yeah, I considered this too. From what I can tell, SqlClient doesn't come with localised versions of the strings, only English.